metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/events/ReferenceSequenceEventHandler.java

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.persistence.events;
+
+import org.springframework.data.rest.core.annotation.HandleBeforeCreate;
+import org.springframework.data.rest.core.annotation.HandleBeforeLinkDelete;
+import org.springframework.data.rest.core.annotation.HandleBeforeLinkSave;
+import org.springframework.data.rest.core.annotation.HandleBeforeSave;
+import org.springframework.data.rest.core.annotation.RepositoryEventHandler;
+import uk.ac.ebi.ampt2d.metadata.exceptionhandling.ReferenceSequenceWithoutValidTaxonomyException;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.ReferenceSequence;
+
+@RepositoryEventHandler(ReferenceSequence.class)
+public class ReferenceSequenceEventHandler {
+
+    @HandleBeforeCreate
+    @HandleBeforeSave
+    public void validateTaxonomies(ReferenceSequence referenceSequence) {
+        validateTaxonomyLink(referenceSequence);
+    }
+
+    @HandleBeforeLinkDelete
+    @HandleBeforeLinkSave
+    public void validateTaxonomies(ReferenceSequence referenceSequence, Object taxonomy) {
+        validateTaxonomyLink(referenceSequence);
+    }
+
+    private void validateTaxonomyLink(ReferenceSequence referenceSequence) {
+        if (referenceSequence.getTaxonomy() == null) {
```

Comment:
This can be checked using @NotNull in the pojo

Reply:
Yes, I and Zhi had discussion on this https://github.com/EBIvariation/metadata-ws/pull/51#discussion_r275762826. 
Should we pass specific message or generic message to user.

Reply:
@NotNull will give u a default exception message that is far more enough I guess.

Reply:
There is are 2 choices
1. @Notnull
This works best when there is no link for taxonomy.
When invalid Taxonomy link is passed, handler will strip-off invalid link and @NotNull will throw error "Taxonomy field can be null".
2. Handle in ExceptionHandlers.java 
Error code can be verified here; pass the message `Null or invalid value is provided` (without Taxonomy word) to user. Since it applies to all such constraints it can not be confined to Taxonomy

Reply:
I, too, feel that adding an event handler for just a linking constraint violation is excessive. Especially given that the error irrecoverable anyway, i. e. there is no scenario where we would want to catch `ReferenceSequenceWithoutValidTaxonomyException` and do something other than informing the user. `@NotNull(message="A reference sequence must have one valid taxonomy")` will do just fine.